Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+        let nodeployArgument = argumentExtractor.extractFlag(named: "nodeploy") > 0
+        let verboseArgument = argumentExtractor.extractFlag(named: "verbose") > 0
+        let productsArgument = argumentExtractor.extractOption(named: "products")
+        let configurationArgument = argumentExtractor.extractOption(named: "configuration")
+
+        // define deployment option
+        self.noDeploy = nodeployArgument
+
+        // define logging verbosity
+        self.verboseLogging = verboseArgument
+
+        // define products
+        self.explicitProducts = !productsArgument.isEmpty
+        if self.explicitProducts {
+            let products = try context.package.products(named: productsArgument)
+            for product in products {
+                guard product is ExecutableProduct else {
+                    throw Errors.invalidArgument(
+                        "product named '\(product.name)' is not an executable product")
+                }
+            }
+            self.products = products
+
+        } else {
+            self.products = context.package.products.filter {
+                $0 is ExecutableProduct && $0.name != "Deploy"
+            }
+        }
+
+        // define build configuration
+        if let buildConfigurationName = configurationArgument.first {
+            guard
+                let buildConfiguration = PackageManager.BuildConfiguration(rawValue: buildConfigurationName)
+            else {
+                throw Errors.invalidArgument(
+                    "invalid build configuration named '\(buildConfigurationName)'")
+            }
+            self.buildConfiguration = buildConfiguration
+        } else {
+            self.buildConfiguration = .release
+        }
+
+        // search for deployment configuration executable
+        let deployProducts = context.package.products.filter { $0.name == "Deploy" }
+        guard deployProducts.count == 1,
+              deployProducts[0].targets.count == 1
+        else {
+            throw Errors.deploymentDescriptorProductNotFound("Deploy")
+        }
+        for t in deployProducts[0].targets {
+            print("\(t.name) - \(t.directory)")
+        }
+        guard deployProducts[0].targets.count == 1 else {
+            fatalError("There is more than one Deploy target \(deployProducts[0].targets)")
+        }
+#if arch(arm64)
+        let arch = "arm64-apple-macosx"
+#else
+        let arch = "x86_64-apple-macosx"
+#endif
+        var deployExecutableURL = URL(fileURLWithPath: deployProducts[0].targets[0].directory.string)
+        // while there is /Sources or /Deploy => remove last path component 
+        while deployExecutableURL.path.contains("/Sources") ||
+              deployExecutableURL.path.contains("/Deploy") {
+
+            deployExecutableURL = deployExecutableURL.deletingLastPathComponent()
+        }
+        self.deployExecutable = deployExecutableURL.appendingPathComponent(".build/\(arch)/\(self.buildConfiguration)/Deploy")
```

Comment:
this seems a bit fragile. what information is it trying to obtain? SwiftPM can return the executables path if needed